### PR TITLE
transition to alpine 3.17 (and to nodejs 18)

### PR DESCRIPTION
**Pull request by bradymiller**
transition to alpine 3.17 (and to nodejs 18)

### CODE REVIEW

File: ccdaservice/oe-blue-button-generate/package.json [hunk 1]
```diff
@@ -34,7 +34,6 @@
     "engines": {
         "node": ">= 0.10.0"
     },
-    "bundleDependencies": true,
```
Comment (bradymiller) [inline ccdaservice/oe-blue-button-generate/package.json:37]: Super odd that for some reason the `npm install` was breaking until i removed the above line. Things still testing well (ie. able to create a CCDA). fyi, here was the error was getting with above line on alpine 3.17 (using npm version 9.1.2):
```
/var/www/localhost/htdocs/openemr/ccdaservice # npm install
npm ERR! Cannot read properties of null (reading 'path')

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-12-19T06_34_03_125Z-debug-0.log
```
and more info from that log mentioned above:
```
...
1019 timing idealTree:node_modules/winston/node_modules/readable-stream Completed in 2ms
1020 timing idealTree:node_modules/winston/node_modules/string_decoder Completed in 0ms
1021 timing idealTree:node_modules/xml2js/node_modules/sax Completed in 0ms
1022 timing idealTree:node_modules/oe-schematron-service/node_modules/send/node_modules/ms Completed in 0ms
1023 timing idealTree:buildDeps Completed in 5818ms
1024 timing idealTree:fixDepFlags Completed in 3ms
1025 timing idealTree Completed in 5835ms
1026 timing reify:loadTrees Completed in 5836ms
1027 timing reify:diffTrees Completed in 3ms
1028 silly reify moves {}
1029 timing reify:retireShallow Completed in 1ms
1030 timing reify:createSparse Completed in 13ms
1031 timing arborist:ctor Completed in 0ms
1032 verbose stack TypeError: Cannot read properties of null (reading 'path')
1032 verbose stack     at PackWalker.gatherBundles (/usr/lib/node_modules/npm/node_modules/npm-packlist/lib/index.js:387:25)
1032 verbose stack     at PackWalker.emit (/usr/lib/node_modules/npm/node_modules/npm-packlist/lib/index.js:160:10)
1032 verbose stack     at PackWalker.then (/usr/lib/node_modules/npm/node_modules/ignore-walk/lib/index.js:145:16)
1032 verbose stack     at PackWalker.emit (node:events:513:28)
1032 verbose stack     at PackWalker.emit (/usr/lib/node_modules/npm/node_modules/ignore-walk/lib/index.js:45:21)
1032 verbose stack     at PackWalker.emit (/usr/lib/node_modules/npm/node_modules/npm-packlist/lib/index.js:155:20)
1032 verbose stack     at PackWalker.then (/usr/lib/node_modules/npm/node_modules/ignore-walk/lib/index.js:145:16)
1032 verbose stack     at PackWalker.emit (node:events:513:28)
1032 verbose stack     at PackWalker.emit (/usr/lib/node_modules/npm/node_modules/ignore-walk/lib/index.js:45:21)
1032 verbose stack     at PackWalker.emit (/usr/lib/node_modules/npm/node_modules/npm-packlist/lib/index.js:155:20)
1033 verbose cwd /var/www/localhost/htdocs/openemr/ccdaservice
1034 verbose Linux 5.19.16-76051916-generic
1035 verbose node v18.12.1
1036 verbose npm  v9.1.2
1037 error Cannot read properties of null (reading 'path')
1038 verbose exit 1
1039 timing npm Completed in 5923ms
1040 verbose unfinished npm timer command:install 1671431643159
1041 verbose unfinished npm timer reify 1671431643163
1042 verbose unfinished npm timer reify:loadBundles 1671431649018
1043 verbose unfinished npm timer reifyNode:node_modules/file 1671431649018
1044 verbose code 1
1045 error A complete log of this run can be found in:
1045 error     /root/.npm/_logs/2022-12-19T06_34_03_125Z-debug-0.log
```

Comment (bradymiller) [reply ccdaservice/oe-blue-button-generate/package.json:37]: @sjpadgett ,
Forgot to mention you on above post. Just making sure you are ok with my change above (testing well, ie. making ccda) and there is no obvious pitfall i am missing.

**Comment by bradymiller** [comment]
super odd inconsistent error is happening in ci build where it looks like the runners do not have write permissions sometimes:
```
Error: file_put_contents(/home/runner/work/openemr/openemr/vendor/composer/autoload_namespaces.php): Failed to open stream: Permission denied
In Filesystem.php line 848:
```
Guessing this is something that will sort itself out (ie. problem with github). Will try it out again tomorrow.

**Comment by bradymiller** [comment]
Looks like it may be a composer/github issue since seeing the following stuff during the composer build when installing packages in the broken builds:
```
    Failed to download omnipay/stripe from dist: The "https://api.github.com/repos/thephpleague/omnipay-stripe/zipball/20812498efedc1079baae5fea96567fefd669105" file could not be written to /home/runner/work/openemr/openemr/vendor/composer/tmp-ed36a76397e0921e6021588cd50e94f3: Failed to open stream: Permission denied
    Now trying to download from source
  - Syncing omnipay/stripe (v3.2.0) into cache
```
(hopefully this is a temporary github/composer issue and not an alpine 3.17 issue)

**Comment by bradymiller** [comment]
finally passing testing :) bringing in, here we come alpine 3.17 and nodejs 18 :)

**Comment by stephenwaite** [comment]
bradys-panda

**Pull request merged**